### ISSUE TITLE
nn.Batchnorm1d input shape notation inconsistency

### ISSUE THREAD

**Issue by kuzand** [pytorch/pytorch#71366]
### 📚 The doc issue

In the _Parameters_ section in the documentation for [nn.Batchnorm1d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html) it says:
`num_features – C from an expected input of size (N, C, L) or L from input of size (N, L)`

Then in the _Shape_ section, we have:
```
Input: (N, C) or (N, C, L)
Output: (N, C) or (N, C, L) (same shape as input)
```

The first description uses (N, L) whereas the second uses (N, C) for the 2D input shape.
Which one is the correct and what does the letters C and L stand for?

### Suggest a potential alternative/fix

There is not clear description of what C and L stand for.

**Comment by jaketae** [comment]
I think `C` stands for channels, and `L` for sequence length. I agree with you that it would be better to keep notations consistent.

**Comment by kuzand** [comment]
So according to the documentation:

`num_features: C from an expected input of size (N, C, L) or L from input of size (N, L)`

we have that `num_features  (number of features) = C (number of channels)` for a 3D input, whereas `num_features  = L (sequence length)` for a 2D input. This is quite confusing because for both a 2D input and a 3D input the last dimension is denoted by the same letter L.

My suggestion would be to use the following notation to avoid any confusion:

```
Input: (N, H_in) or (N, C, L)
Output: (N, H_in) or (N, C, L) (same shape as input)
```

where N is the batch size, H_in is the number of features (similar notation as in nn.Linear), C is the number of channels (which is equal to the number of features), L is the length of sequence.

**Comment by kuzand** [comment]
Another option would be to use the letter C in both cases:

`num_features: number of features or channels`
```
Input: (N, C) or (N, C, L)
Output: (N, C) or (N, C, L) (same shape as input)
```
